### PR TITLE
Final consensus status processing

### PR DESCRIPTION
**Pull request by atif-konasl**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**
It gets the status using notification by processOrcConfirmationLoop and then it takes action based on status of block. If status is-
**Verified:**
- return nil

**Invalid:**
- directly return error and discard the pending block.
- sync service will re-download the block

**Pending:**
- Re-check new response from orchestrator
- Decrease the re-try limit if it gets pending status again
- If it reaches the maximum limit then return error

**Which issues(s) does this PR fix?**
Issue ID: [22](https://github.com/lukso-network/vanguard-consensus-engine/issues/22)

**Other notes for review**
Added two new flags(`--lukso-network` and `--orc-http-provider`) for identifying vanguard node.  So `--lukso-network` defines a bool flag which is required to run vanguard node and ` --orc-http-provider` flag contain endpoint of orchestrator client for example: `--orc-http-provider=http://127.0.0.1:8545`

### CODE REVIEW

**Comment by atif-konasl** [comment]
@mxmar I need to pull your changes.

File: beacon-chain/blockchain/BUILD.bazel [hunk 1]
```diff
@@ -126,11 +125,11 @@ go_test(
         "@com_github_prysmaticlabs_ethereumapis//eth/v1alpha1:go_default_library",
         "@com_github_sirupsen_logrus//:go_default_library",
         "@com_github_sirupsen_logrus//hooks/test:go_default_library",
-        "@com_github_gogo_protobuf//types:go_default_library",
         "@in_gopkg_d4l3k_messagediff_v1//:go_default_library",
         "@org_golang_x_net//context:go_default_library",
-        "@org_golang_google_grpc//codes:go_default_library",
-        "@org_golang_google_grpc//status:go_default_library",
+        "//beacon-chain/core/feed/block:go_default_library",
```
Comment (mxmar) [inline beacon-chain/blockchain/BUILD.bazel:130]: I think you don't need to import this because it's included already as `embed = [":go_default_library"],` in your `go_test`

File: beacon-chain/blockchain/testing/mock.go [hunk 2]
```diff
@@ -383,6 +417,6 @@ func (s *ChainService) VerifyFinalizedConsistency(_ context.Context, r []byte) e
 }
 
 // Vanguard: UnConfirmedBlocksFromCache mocks UnConfirmedBlocksFromCache method and send it nil
-func (s *ChainService) UnConfirmedBlocksFromCache() ([]*ethpb.BeaconBlock, error) {
+func (s *ChainService) SortedUnConfirmedBlocksFromCache() ([]*ethpb.BeaconBlock, error) {
```
Comment (mxmar) [inline beacon-chain/blockchain/testing/mock.go:420]: Just naming convention -> `Unconfirmed`

File: beacon-chain/blockchain/van_process_pending_blocks.go [hunk 3]
```diff
@@ -58,10 +78,275 @@ func (s *Service) publishAndStorePendingBlockBatch(ctx context.Context, pendingB
 }
 
 // UnConfirmedBlocksFromCache retrieves all the cached blocks from cache and send it back to event api
-func (s *Service) UnConfirmedBlocksFromCache() ([]*ethpb.BeaconBlock, error) {
+func (s *Service) SortedUnConfirmedBlocksFromCache() ([]*ethpb.BeaconBlock, error) {
```
Comment (mxmar) [inline beacon-chain/blockchain/van_process_pending_blocks.go:74]: Same here -> `SortedUnconfirmedBlocksFromCache`

File: beacon-chain/orchestrator/BUILD.bazel [hunk 4]
```diff
@@ -2,7 +2,7 @@ load("@prysm//tools/go:def.bzl", "go_library")
 load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
 go_library(
-    name = "orchestrator",
+    name = "go_default_library",
```
Comment (mxmar) [inline beacon-chain/orchestrator/BUILD.bazel:5]: This is because of `prysm` project convention? (they are using `dafault` naming in this project)

File: beacon-chain/rpc/beacon/van_pending_blocks.go [hunk 5]
```diff
@@ -19,7 +19,7 @@ func (bs *Server) StreamNewPendingBlocks(empty *ptypes.Empty, stream ethpb.Beaco
 	defer unConfirmedblockSub.Unsubscribe()
 
 	// Getting un-confirmed blocks from cache and sends those blocks to orchestrator
-	unconfirmedBlocks, err := bs.UnconfirmedBlockFetcher.UnConfirmedBlocksFromCache()
+	unconfirmedBlocks, err := bs.UnconfirmedBlockFetcher.SortedUnConfirmedBlocksFromCache()
```
Comment (mxmar) [inline beacon-chain/rpc/beacon/van_pending_blocks.go:22]: lower case -> `SortedUnconfirmedBlocksFromCache`

File: beacon-chain/core/feed/block/events.go [hunk 6]
```diff
@@ -3,22 +3,33 @@
...
+	ConfirmedBlock
 )
 
 // ReceivedBlockData is the data sent with ReceivedBlock events.
 type ReceivedBlockData struct {
 	SignedBlock *ethpb.SignedBeaconBlock
 }
 
-//
+// UnConfirmedBlockData is the data sent to orchestrator
```
Comment (mxmar) [inline beacon-chain/core/feed/block/events.go:25]: lower case for `UnconfirmedBlockData

**Review by mxmar — APPROVED** [review]
Code looks good to me in general. These things are new to `beacon-chain` client so it's hard to say looking at the code if BE logic will cover all edge cases and our goals. Nice `table testing` - that's great that we have `go` tests for this implementation 👍 

GH actions are green so it's possible to check, build and test this codebase by `go` (not only by `bazel`)

File: beacon-chain/blockchain/testing/mock.go (hunk 2 above)
Comment (atif-konasl) [reply beacon-chain/blockchain/testing/mock.go:420]: Thanks right.

File: beacon-chain/blockchain/van_process_pending_blocks.go (hunk 3 above)
Comment (atif-konasl) [reply beacon-chain/blockchain/van_process_pending_blocks.go:74]: Thanks right :)

File: beacon-chain/orchestrator/BUILD.bazel (hunk 4 above)
Comment (atif-konasl) [reply beacon-chain/orchestrator/BUILD.bazel:5]: Yes. This is a module and all module should have same conventional name.
